packages/internals/src/__tests__/engine-commands/queryEngineCommons.test.ts

```diff
@@ -54,6 +58,32 @@ describeIf(process.env.PRISMA_CLI_QUERY_ENGINE_TYPE !== 'binary')('loadNodeAPILi
     }
   })
 
+  it('error path, architecture or libc', async () => {
+    const spyLoadTag = `Unable to require(\`/app/node_modules/.pnpm/prisma@4.0.0/node_modules/prisma/libquery_engine-linux-arm64-openssl-1.1.x.so.node\`)
+Error relocating /app/node_modules/.pnpm/prisma@4.0.0/node_modules/prisma/libquery_engine-linux-arm64-openssl-1.1.x.so.node: __res_init: symbol not found`
+
+    const spyLoad = jest.spyOn(loadUtils, 'load').mockImplementation((id: string) => {
+      throw new Error(spyLoadTag)
+    })
```

Comment:
I think this will cause problems like in https://github.com/prisma/schema-team/issues/360 which means we can never upgrade the `@swc/core` dependency of the internals package...

We really need to find a solution (didn't find any) or test differently